components/needs_team/needs_team.jsx

```diff
@@ -205,6 +207,11 @@ export default class NeedsTeam extends React.Component {
         this.props.actions.loadStatusesForChannelAndSidebar();
         this.props.actions.loadProfilesForDirect();
 
+        this.props.actions.getAllGroupsAssociatedToChannelsInTeam(team.id);
+        if (team.group_constrained) {
+            this.props.actions.getAllGroupsAssociatedToTeam(team.id);
+        }
```

Comment:
need to add getGroups() call for the case where neither the team nor the channels in the team are synced - could be expensive to do it for every team switch.